functests/dpdk/dpdk.go

```diff
@@ -144,135 +126,204 @@ var _ = Describe("dpdk", func() {
+				Expect(activeNumberOfFreeHugePages).To(BeNumerically(">", 0))
 
-			// In case of nodeselector set, this pod will end up in a compliant node because the selection
-			// logic is applied to the workload pod.
-			pod := pods.DefineWithHugePages(namespaces.DpdkTest, dpdkWorkloadPod.Spec.NodeName)
-			pod, err := client.Client.Pods(namespaces.DpdkTest).Create(context.Background(), pod, metav1.CreateOptions{})
-			Expect(err).ToNot(HaveOccurred())
+				// In case of nodeselector set, this pod will end up in a compliant node because the selection
+				// logic is applied to the workload pod.
+				pod := pods.DefineWithHugePages(namespaces.DpdkTest, dpdkWorkloadPod.Spec.NodeName)
+				pod, err := client.Client.Pods(namespaces.DpdkTest).Create(context.Background(), pod, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() int {
-				buff, err := pods.ExecCommand(client.Client, *dpdkWorkloadPod, []string{"cat",
-					fmt.Sprintf("/sys/devices/system/node/node%d/hugepages/hugepages-1048576kB/free_hugepages", numaNode)})
+				Eventually(func() int {
+					buff, err := pods.ExecCommand(client.Client, *dpdkWorkloadPod, []string{"cat",
+						fmt.Sprintf("/sys/devices/system/node/node%d/hugepages/hugepages-1048576kB/free_hugepages", numaNode)})
+					Expect(err).ToNot(HaveOccurred())
+					numberOfFreeHugePages, err := strconv.Atoi(strings.Replace(buff.String(), "\r\n", "", 1))
+					Expect(err).ToNot(HaveOccurred())
+					return numberOfFreeHugePages
+				}, 5*time.Minute, 5*time.Second).Should(Equal(activeNumberOfFreeHugePages - 1))
+
+				pod, err = client.Client.Pods(namespaces.DpdkTest).Get(context.Background(), pod.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				numberOfFreeHugePages, err := strconv.Atoi(strings.Replace(buff.String(), "\r\n", "", 1))
+				Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+
+				err = client.Client.Pods(namespaces.DpdkTest).Delete(context.Background(), pod.Name, metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64Ptr(0)})
 				Expect(err).ToNot(HaveOccurred())
 
-				return numberOfFreeHugePages
-			}, 5*time.Minute, 5*time.Second).Should(Equal(activeNumberOfFreeHugePages - 1))
+				Eventually(func() error {
+					_, err := client.Client.Pods(namespaces.DpdkTest).Get(context.Background(), pod.Name, metav1.GetOptions{})
+					if err != nil && errors.IsNotFound(err) {
+						return err
+					}
+					return nil
+				}, 10*time.Second, 1*time.Second).Should(HaveOccurred())
+			})
+		})
+	})
 
-			pod, err = client.Client.Pods(namespaces.DpdkTest).Get(context.Background(), pod.Name, metav1.GetOptions{})
+	Context("VFS split for dpdk and netdevice", func() {
+		BeforeEach(func() {
+			if discovery.Enabled() {
+				Skip("Split VF test disabled for discovery mode")
+			}
+		})
+		execute.BeforeAll(func() {
+			createSriovPolicyAndNetworkShared()
+			var err error
+			dpdkWorkloadPod, err = createDPDKWorkload(nodeSelector,
```

Comment:
should you clean the old pods before?

Reply:
That was done inside `createSriovPolicyAndNetworkShared` / `createSriovPolicyAndNetworkDPDKOnly` (as it was before).
Will move the clean function outside, I agree it will improve readability.